Parsing and visualization example for csv data from logistic forecast

ashander opened the pull request:
a step toward the overall goal of providing small example of parsing and visualizing data in the standard format(s). for now this is the csv format for the simplest example forecast.

Commit:
Example of working with and visuzlizing csv

ashander commented:
Note this contains the commits in #21

mdietze commented on vignettes/visualization-example.Rmd:
```diff
@@ -0,0 +1,155 @@
+
+## Reading structured methods
+
+Load the additional metadata (link to  defs of additional metadata or paste here?)
```
I would just say "see documentation" rather than linking to the documentation because the doc link changes frequently (currently in gDocs rather than Rmd files within the package itself -- the latter would be better but I don't have the time/bandwidth to translate and gdoc has been easier for group discussions)

mdietze commented on vignettes/visualization-example.Rmd:
````diff
@@ -0,0 +1,155 @@
+
+Details about the number of state variables and parameters, and the propagation of uncertainty. For example, details of the treatment of process error in this model are given by
+```{r process}
+methods_md %>% eml_get("process_error") # note we cold do this directly on \`md` object
````
type on `cold`

mdietze reviewed (commented):
There were a bunch of pkgdown things in this PR that seem to be redundant with #21, but I suspect/hope that there won't be conflicts

ashander commented:
> There were a bunch of pkgdown things in this PR that seem to be redundant with #21, but I suspect/hope that there won't be conflicts

Changes make sense. Re #21 yes I wrote this on top of that one because I needed to update the eml file so it had two spp etc. I will rebase onto that one after making the changes.

Commit:
Fixes to language

Commit:
Fix to logistic so doesn't write .nc

Commit:
CLarify outstanding questions

ashander commented:
@mdietze before any merge there are a few outstanding questions (marked now by TODO). if it looks good I can update the pkgdown site again (I removed those commits from this rebase)

I've also

- fixed some spelling errors and clarified the language a bit in places:  d381071
- I also made a fix to the logistic example here (to avoid writing .nc when I rebuilt).

mdietze commented on vignettes/logistic-metadata-example.Rmd:
```diff
@@ -543,6 +543,8 @@ emlname <- system.file("extdata", "forecast-eml.xml", package="EFIstandards")
 emldname <- system.file("extdata", "forecast-eml.json", package="EFIstandards")
 emld::as_json(as_emld(emlname), file = emldname)
+ncname <- system.file("extdata", ncfname, package="EFIstandards")
+# TODO  -- add writing the nc file here too
```
👍

mdietze commented on vignettes/visualization-example.Rmd:
````diff
@@ -0,0 +1,168 @@
+n_ensembles <- n_ensembles[[1]]
+```
+
+From this metadata, the output of this forecast should have error propagation via an ensemble of size `r n_ensembles`, two dimensions of process error (TODO one for each state variable? or two parameters govern it?) with no covariance between these. Further, one can check if the forecast used data assimilation (no):
````
In this particular case the $\sigma$ for the two state variables happen to be the same.

mdietze commented on vignettes/visualization-example.Rmd:
```diff
@@ -0,0 +1,168 @@
+
+Cross-referencing this documentation of EFI forecasting standards, note that
+`data_assimilation` and `forecast` columns are zeroed out so this was a
+hindcast (TODO?) and performed no data assimilation. As expected, there are `r
```
Not sure what the TODO is here, but yes this was a hindcast

mdietze reviewed (approved):
responded to the TODOs I could find

Commit:
Fixes

Commit:
Pkgdown site update

Commit:
Update logistic to write nc if needed

ashander commented:
Alright I fixed all that stuff. Thanks
